mllib/src/main/scala/org/apache/spark/ml/regression/LinearRegression.scala

```diff
@@ -298,11 +298,7 @@ class LinearRegressionModel private[ml] (
    */
   // TODO: decide on a good name before exposing to public API
   private[regression] def evaluate(dataset: DataFrame): LinearRegressionSummary = {
-    val t = udf { features: Vector => predict(features) }
-    val predictionAndObservations = dataset
-      .select(col($(labelCol)), t(col($(featuresCol))).as($(predictionCol)))
-
-    new LinearRegressionSummary(predictionAndObservations, $(predictionCol), $(labelCol))
+    new LinearRegressionSummary(transform(dataset), $(predictionCol), $(labelCol))
```

Comment:
I'm not really sure handling linear regression without a prediction column makes a lot of sense, or even explicitly checking the predictionCol here. The previous version assumed it was set, as do many of the other transformations (it also has a default value of "prediction"). I think if the user intentionally clears the prediction column or sets it to an invalid column it would be better to stick with the existing behaviour and messaging for consistency across the methods. That being said I will of course defer to your judgement.